modules/post/windows/gather/enum_chocolatey_applications.rb

```diff
@@ -0,0 +1,69 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::Windows::Powershell
+
+  def initialize(info = {})
+    super(update_info(info,
+                      "Name" => "Windows Gather Installed Application Within Chocolatey Enumeration",
+                      "Description" => %q{ This module will enumerate all installed applications on a Windows system with chocolatey installed },
+                      "License" => MSF_LICENSE,
+                      "Author" => ["Nick Cottrell <ncottrellweb@gmail.com>"],
+                      "Platform" => ["win"],
+                      "SessionTypes" => ["meterpreter", "shell"]))
+  end
+
+  def have_chocolatey?
+    cmd_exec("choco.exe", "-v") =~ /\d+\.\d+\.\d+/m
+  end
+
+  def run
+    # checking that session is meterpreter and session has powershell
+    return 0 if !(have_powershell? || have_chocolatey?)
+    if session.type == "meterpreter"
+      print_status("Enumerating applications installed on #{sysinfo["Computer"]}")
+    end
+
+    # getting chocolatey version
+    choco_version = cmd_exec("choco", "-v")
+    print_status("Targets Chocolatey version: #{choco_version}")
+
+    # Getting results of listing chocolatey applications
+    print_status("Getting chocolatey applications.")
+
+    # checking if chocolatey is 2+ or 1.0.0
+    if choco_version.match(/^1\.\d+\.\d+$/)
+      # its version 1, use local only
+      data = cmd_exec("choco", "list -lo")
+    else
+      # its version 2 or above, no need for local
+      data = cmd_exec("choco", "list")
```

Comment:
Might always have to pass `-lo` actually because gapped systems like this meterp build host can't get to the wider world:
```
PS C:\Users\Administrator> choco list
Chocolatey v0.12.1
[NuGet] Not able to contact source 'https://community.chocolatey.org/api/v2/'. Error was The remote name could not be resolved: 'community.chocolatey.org'
[NuGet] Not able to contact source 'https://community.chocolatey.org/api/v2/'. Error was The remote name could not be resolved: 'community.chocolatey.org'
0 packages found.
PS C:\Users\Administrator> choco list -lo
Chocolatey v0.12.1
chocolatey 0.12.1
chocolatey-core.extension 1.3.5.1
chocolatey-dotnetfx.extension 1.0.1
chocolatey-visualstudio.extension 1.10.0
chocolatey-windowsupdate.extension 1.0.4
DotNet4.6.1 4.6.01055.20170308
dotnetfx 4.8.0.20190930
git 2.35.1.2
git.install 2.35.1.2
git.portable 2.35.1.2
gitextensions 3.5.4
KB2919355 1.0.20160915
KB2919442 1.0.20160915
KB2999226 1.0.20181019
KB3033929 1.0.5
KB3035131 1.0.3
patch 2.5.9
visualstudio-installer 2.0.2
visualstudio2019community 16.11.11.0
windowsdriverkit10 10.0.19041.685
20 packages installed.
PS C:\Users\Administrator>
```

Reply:
I did a better resolution for this in commit `6ef3cf7`. I wanted to keep the loss of `-lo` for later versions to ensure that the module wouldn't break for future versions in case they were to remove the argument in future versions afterwards